File: LeetCode/0026.remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (0 == nums.size())   return 0;
        int size = nums.size();
        int pos = 1, index = 1;
        while (index < size){
            if (nums[index] != nums[index - 1]){
                nums[pos] = nums[index];
                pos++;
                index++;
                continue;
            }
            while (index < size && nums[index] == nums[index - 1])  index++;
        }
        return pos;
    }
};

File: LeetCode/0026.remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (null == nums || 0 == nums.length)   return 0;
        int length = nums.length;
        int pos = 1;
        int index = 1;
        while (index < length){
            if (nums[index] != nums[index - 1]){
                if (index != pos){
                    nums[pos] = nums[index];
                }
                index++;
                pos++;
                continue;
            }
            while (index < length && nums[index] == nums[index - 1])  index++;

        }
        return pos;
    }
}